### DISCUSSION TITLE
Problem with updating Json field in Postgres DB

### DISCUSSION THREAD

**Discussion by Darent98** [mikro-orm/mikro-orm#5356] [Q&A] [answered]
Hi, 

I don't know is this is a bug or a misconfiguration on my side, but when I try to update the whole json stored in a JsonType field with assign, it is merged instead of overwritten. I could not find a setting, that just overwrites the value in db with the new json I set. 
Because of this keys of my object, that are not defined anymore, stay in the json instead of getting removed.

At the end is a minimal reproduction example for the reproduction container.

I would love any suggestions on how to achive to overwrite the whole json.

``` ts
import { Entity, JsonType, MikroORM, PostgreSqlDriver, PrimaryKey, Property } from '@mikro-orm/postgresql';

@Entity()
class Test {

  @PrimaryKey()
  id!: number;

  @Property({type: JsonType, nullable: true})
    data!: any

  constructor(data: any) {
    this.data = data
  }

}

let orm: MikroORM;

beforeAll(async () => {
  orm = await MikroORM.init<PostgreSqlDriver>({
    dbName: 'test',
    user: 'test',
    password: 'test',
    entities: [Test],
    debug: ['query', 'query-params'],
    allowGlobalContext: true, // only for testing
  });
  await orm.schema.refreshDatabase();
});

afterAll(async () => {
  await orm.close(true);
});

test('basic CRUD example', async () => {
  const testData = {'a': {'attr': 1}, 'b': {'attr': 2}}
  orm.em.create(Test, { 
    data: testData 
   });
  await orm.em.flush();
  orm.em.clear();

  const test = (await orm.em.findAll(Test, {limit: 1}))[0];
  expect(test.data).toStrictEqual(testData);

  const testData2 = {'a': {'attr': 3}}
  orm.em.assign(test, {id: test.id, data: testData2});
  await orm.em.flush();

  expect(test.data).toStrictEqual(testData2); // fails, because 'b' still exists in data
  orm.em.remove(test)
  await orm.em.flush()
});
```

**Comment by Darent98** [comment, marked as answer]
Okay, nevermind. After a bit more research I tried the `mergeObjectProperties` option and set it to false. With this the test passes and the object is not merged. Before I only tried it with `merge: false` which did not help.

**Reply by B4nan** [reply]
> Before I only tried it with merge: false which did not help.

That controls something else - it means that when an object is in place of a relation, it is considered as an existing entity instead of a new one - so internally `em.merge()` is called on it instead of `em.create()`.
